catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderRealTimeQueryTest.java

```diff
@@ -261,4 +257,139 @@ public void testRealTimeQueryNotId() throws Exception {
     // Verify a "not ID query" does not return the result.
     queryAndVerifyCount(0, filter, provider);
   }
+
+  /*similar to testRealTimeQueryNotId but this time we'll use a random ID for the filter. In this case
+  the created metacard will match the filter but should not be returned because we shouldn't be doing a
+  real time query.*/
+  @Test
+  public void testRealTimeQueryNotId2() throws Exception {
+
+    final String metacardTitle = "testRealTimeQueryNotId";
+
+    deleteAll(provider);
+
+    MockMetacard metacard = new MockMetacard(Library.getFlagstaffRecord());
+    metacard.setTitle(metacardTitle);
+
+    CreateResponse createResponse = create(metacard, provider);
+    Optional<String> id =
+        createResponse.getCreatedMetacards().stream().map(m -> m.getId()).findFirst();
+
+    if (!id.isPresent()) {
+      Assert.fail("Metacard creation failed for metacard with title: " + metacardTitle);
+    }
+
+    // Verify the result is actually visible (searchable) in Solr using a real time query by ID.
+    queryAndVerifyCount(
+        1, getFilterBuilder().attribute(Metacard.ID).is().equalTo().text(id.get()), provider);
+
+    Filter filter =
+        getFilterBuilder()
+            .not(getFilterBuilder().attribute(Metacard.ID).equalTo().text("can't match this"));
+
+    // Verify a "not ID query" does not return the result.
+    queryAndVerifyCount(0, filter, provider);
+  }
+
+  @Test
+  public void testRealTimeQueryByIdOrTitle() throws Exception {
+
+    final String metacardTitle = "testRealTimeQueryById";
+
+    deleteAll(provider);
+
+    MockMetacard metacard = new MockMetacard(Library.getFlagstaffRecord());
+    metacard.setTitle(metacardTitle);
+
+    CreateResponse createResponse = create(metacard, provider);
+    Optional<String> id =
+        createResponse.getCreatedMetacards().stream().map(m -> m.getId()).findFirst();
+
+    if (!id.isPresent()) {
+      Assert.fail("Metacard creation failed for metacard with title: " + metacardTitle);
+    }
+
+    // Real time queries only work when querying by ID, so a real time query by title only will
```

Comment:
✏️ It took me a while to realize what this test was trying to do.  Might want to make it clear that a real time get by ID will not be done since it is ORed with a non-ID part of the query.  The current wording make it sound like a real time get will be done which would be able to return the ID if true.